lib/events/pgevents/pgevents.go

```diff
@@ -0,0 +1,593 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pgevents
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/gravitational/trace"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/sirupsen/logrus"
+
+	"github.com/gravitational/teleport/api/types"
+	apievents "github.com/gravitational/teleport/api/types/events"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/events"
+	pgcommon "github.com/gravitational/teleport/lib/events/pgevents/common"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+const (
+	Schema    = "postgres"
+	AltSchema = "postgresql"
+
+	// componentName is the component name used for logging.
+	componentName = "pgevents"
+)
+
+const (
+	defaultRetentionPeriod = 8766 * time.Hour // 365.25 days
```

Comment:
Casually also called a year :sweat_smile:

Reply:
updated in 445c9d7f8d 🤔